src/components/EmbeddedActivity.tsx

```diff
@@ -82,8 +82,8 @@ export default function EmbeddedActivity({ participant, activity, name, onComple
 
   const activateEmbeddedActivity = async (activity) => {
     setActivityId(activity.id)
-    activity.settings.language = i18n.language
-    setSettings(activity.settings)
+    setSaved(false)   
+    setSettings({ ...settings, settings: activity.settings, configuration: { language: i18n.language } })
```

Comment:
I imagine `...settings` is a backwards-compatibility thing that will be removed later?

Reply:
I don't really understand the question. Why it is to be removed?

Reply:
Sorry, I don't mean it needs to be removed, but just trying to understand whether that `...settings` is there for backwards compatibility purposes or some other reason. The code looks good to go though!